index.js

```diff
@@ -94,6 +97,32 @@ Electron.prototype._spawn = function(url){
       case 'stderr': self.stderr.write(msg[1]); break;
     }
   });
+
+  // these event callbacks below will display the output and/or error 
+  // streams from an optional background process like Xvfb when running
+  // headless in a container, otherwise electron can fail silently
+  const errs = [], outs = [];
+  ps.stdout.on('data', function(data) {
+    outs.push(data)
+  });
+
+  ps.stderr.on('data', function(data) {
+    errs.push(data)
+  });
+
+  ps.on('close', function(code) {
+    const err = errs.join('').trim()
+    if (err.length) {
+      console.error(err)
```

Comment:
It's best practice for a node library never to write to stdout/err, unless it's required. Since this is a nodejs stream, you can do `this.emit('error', new Error(err.join('').trim())`, this would be the standard way to pass errors to the outside.